altos-rust/altos-core/src/sched/mod.rs

```diff
@@ -24,12 +25,13 @@ pub static mut CURRENT_TASK: Option<Box<Node<TaskControl>>> = None;
 
 pub static PRIORITY_QUEUES: [SyncQueue<TaskControl>; NUM_PRIORITIES] = [SyncQueue::new(),
                                                                     SyncQueue::new(), 
+                                                                    SyncQueue::new(), 
                                                                     SyncQueue::new()];
 pub static SLEEP_QUEUE: SyncQueue<TaskControl> = SyncQueue::new();
 pub static DELAY_QUEUE: SyncQueue<TaskControl> = SyncQueue::new();
 pub static OVERFLOW_DELAY_QUEUE: SyncQueue<TaskControl> = SyncQueue::new();
-static mut CRITICAL_TASK_COUNTER: u8 = 0; //Number of Critical Tasks that have run
-const CRITICAL_TASK_MAX: u8 = 10; //Maximum number of critical tasks that can run
+pub static CRITICAL_TASK_COUNTER: AtomicUsize = AtomicUsize::new(0);//Number of Critical Tasks that have run
```

Comment:
Use `ATOMIC_USIZE_INIT` instead of `AtomicUsize::new(0)` for compatibility reasons